ui/test/nuxeo-layout.test.js

```diff
@@ -27,57 +35,103 @@ const base = url.substring(0, url.lastIndexOf('/'));
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href="notfound.html"></nuxeo-layout>
+        `,
+      );
+      await flush();
 
-  test('layout stamped with model', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
-    `);
-    await waitChanged(layout, 'element');
-    expect(layout.$$('nuxeo-error').hidden).to.be.true;
-    expect(layout.element.localName).to.equal('dummy-layout');
-    expect(layout.element.text).to.equal('dummy');
-  });
+      const nuxeoError = layout.$$('nuxeo-error');
+      if (!isElementVisible(nuxeoError)) {
+        await waitForAttrMutation(nuxeoError, 'hidden', null);
+      }
+      expect(isElementVisible(layout.$$('#container'))).to.be.false;
+      expect(isElementVisible(nuxeoError)).to.be.true;
+    });
 
-  test('layout valid field invalid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
-  });
+    test('Should display error messages when layout is not found', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href="notfound.html"></nuxeo-layout>
+        `,
+      );
+      await flush();
 
-  test('layout valid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "valid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(true);
+      const nuxeoError = layout.$$('nuxeo-error');
+      expect(nuxeoError.code).to.equal('404');
+      expect(nuxeoError.url).to.equal('notfound.html');
+      expect(nuxeoError.message).to.equal('Failed to find layout');
+    });
+
+    test('Should hide layout when href is empty', async () => {
+      const layout = await fixture(
+        html`
+          <nuxeo-layout href=""></nuxeo-layout>
+        `,
+      );
+      expect(isElementVisible(layout)).to.be.false;
+    });
   });
 
-  test('layout invalid field valid', async () => {
-    const layout = await fixture(html`
-      <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "invalid", "required": "foo"}'></nuxeo-layout>
-    `);
-    const validity = await layout.validate();
-    expect(validity).to.equal(false);
+  suite('Stamping', () => {
+    test('Should stamp the layout when model is defined', async () => {
+      const layout = await fixture(html`
+        <nuxeo-layout href="${base}/dummy-layout.html" model='{"text": "dummy"}'></nuxeo-layout>
+      `);
+      await flush();
+
+      if (!layout.element) {
+        await waitChanged(layout, 'element');
+      }
+
+      const span = layout.element.$$('span');
```

Comment:
We should check that the span exists before checking its innerText. And also that the other widget (`nuxeo-input`) was stamped as well, to the sake of layout integrity.